### PR TITLE
Optimize updateLatestPoints

### PR DESCRIPTION
**Pull request by NullSoldier**
## Summary

This turns 3 queries into 1

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: src/events/events.service.ts [hunk 1]
```diff
@@ -596,46 +596,36 @@ export class EventsService {
...
+    const [counts] = await this.prisma.readClient.$queryRawUnsafe<
+      {
+        count: BigInt;
+        points: BigInt;
+        total_points: BigInt;
+        last_occurred_at: Date;
+      }[]
+    >(
+      `select
+        max(occurred_at) as last_occurred_at,
```
Comment (NullSoldier) [inline src/events/events.service.ts:608]: this needs to be filtered on the type

File: src/events/events.service.ts [hunk 2]
```diff
@@ -596,46 +596,36 @@ export class EventsService {
...
+        sum(points) as total_points,
+        count(*) as total_count,
+        sum(case when type = $1::event_type then points end) points,
+        count(case when type = $1::event_type then 0 end) as count
```
Comment (NullSoldier) [inline src/events/events.service.ts:612]: then 1

File: src/events/events.service.ts (hunk 1 above)
Comment (NullSoldier) [reply src/events/events.service.ts]: fixed

File: src/events/events.service.ts (hunk 2 above)
Comment (NullSoldier) [reply src/events/events.service.ts]: fixed

File: src/events/events.service.ts [hunk 3]
```diff
@@ -596,46 +596,36 @@ export class EventsService {
...
-        points: true,
-      },
-      where: {
-        user_id: userId,
-      },
-    });
-    const totalPoints = totalPointsAggregate._sum.points ?? 0;
+    const [counts] = await this.prisma.readClient.$queryRawUnsafe<
+      {
+        count: BigInt;
```
Comment (jowparks) [inline src/events/events.service.ts:601]: Looks like you are only using 4 of the 5 selected values? Also I take it ordering doesn't matter?

**Review by jowparks — APPROVED** [review]

File: src/events/events.service.ts (hunk 3 above)
Comment (NullSoldier) [reply src/events/events.service.ts:601]: yeah, order of keys in JS doesn't matter.

Comment (NullSoldier) [reply src/events/events.service.ts:601]: Removed total_count.

**Pull request merged**